test/units/modules/network/aci/test_aci_rest.py

```diff
@@ -0,0 +1,299 @@
+        json_result = aci_response(json_response, 'json')
+
+        # Python 2.7+ is needed for xmljson
+        if sys.version_info < (2, 7):
+            return
+
+        xml_response = '<?xml version="1.0" encoding="UTF-8"?><imdata totalCount="1">\n<aaaLogin token="ZldYAsoO9d0FfAQM8xaEVWvQPSOYwpnqzhwpIC1r4MaToknJjlIuAt9+TvXqrZ8lWYIGPj6VnZkWiS8nJfaiaX/AyrdD35jsSxiP3zydh+849xym7ALCw/fFNsc7b5ik1HaMuSUtdrN8fmCEUy7Pq/QNpGEqkE8m7HaxAuHpmvXgtdW1bA+KKJu2zY1c/tem" siteFingerprint="NdxD72K/uXaUK0wn" refreshTimeoutSeconds="600" maximumLifetimeSeconds="86400" guiIdleTimeoutSeconds="1200" restTimeoutSeconds="90" creationTime="1500134817" firstLoginTime="1500134817" userName="admin" remoteUser="false" unixUserId="15374" sessionId="o7hObsqNTfCmDGcZI5c4ng==" lastName="" firstName="" version="2.0(2f)" buildTime="Sat Aug 20 23:07:07 PDT 2016" node="topology/pod-1/node-1">\n<aaaUserDomain name="all" rolesR="admin" rolesW="admin">\n<aaaReadRoles/>\n<aaaWriteRoles>\n<role name="admin"/>\n</aaaWriteRoles>\n</aaaUserDomain>\n<DnDomainMapEntry dn="uni/tn-common" readPrivileges="admin" writePrivileges="admin"/>\n<DnDomainMapEntry dn="uni/tn-infra" readPrivileges="admin" writePrivileges="admin"/>\n<DnDomainMapEntry dn="uni/tn-mgmt" readPrivileges="admin" writePrivileges="admin"/>\n</aaaLogin></imdata>\n'''  # NOQA
+        xml_result = aci_response(xml_response, 'xml')
+
+        self.assertEqual(expected_result, json_result)
+        self.assertEqual(json_result, xml_result)
+
+    def test_invalid_input(self):
+        self.maxDiff = None
+
+        expected_result = {
+            u'error_code': u'401',
+            u'error_text': u'Username or password is incorrect - FAILED local authentication',
+            u'imdata': [{
+                u'error': {
+                    u'attributes': {
+                        u'code': u'401',
+                        u'text': u'Username or password is incorrect - FAILED local authentication',
+                    },
+                },
+            }],
+            u'totalCount': '1',
+        }
+
+        json_response = '{"totalCount":"1","imdata":[{"error":{"attributes":{"code":"401","text":"Username or password is incorrect - FAILED local authentication"}}}]}'  # NOQA
+        json_result = aci_response(json_response, 'json')
+
+        # Python 2.7+ is needed for xmljson
+        if sys.version_info < (2, 7):
+            return
+
+        xml_response = '''<?xml version="1.0" encoding="UTF-8"?><imdata totalCount="1">
+        <error code="401" text="Username or password is incorrect - FAILED local authentication"/>
+        </imdata>
+        '''
+        xml_result = aci_response(xml_response, 'xml')
+
+        self.assertEqual(expected_result, json_result)
+        self.assertEqual(json_result, xml_result)
+
+    def test_empty_response(self):
+        self.maxDiffi = None
+
+        if sys.version_info < (3, 0):
```

Comment:
Another test that I would move into version specific classes with pytest skipifs